nextcord/automod.py

```diff
@@ -0,0 +1,333 @@
+
+    @overload
+    async def edit(
+        self,
+        *,
+        name: Optional[str] = ...,
+        event_type: Optional[EventType] = ...,
+        preset: Optional[KeywordPresetType] = ...,
+        notify_channel: Optional[GuildChannel] = ...,
+        timeout_seconds: Optional[int] = ...,
+        enabled: Optional[bool] = ...,
+        exempt_roles: Optional[List[Role]] = ...,
+        exempt_channels: Optional[List[GuildChannel]] = ...,
+    ):
+        ...
+
+    async def edit(self, **fields):
+        """
+        |coro|
+        Edit this auto moderation rule.
+
+        You must have the :attr:`~Permissions.manage_guild` to be able to do this.
+
+        Parameters
+        -----------
+        name: Optional[:class:`str`]
+            The new name of this auto moderation rule.
+        event_type: Optional[:class:`EventType`]
+            The new trigger event type of this auto moderation rule.
+        keyword_filters: Optional[List[:class:`str`]]
+            The keywords that the filter should match.
+
+            .. note::
+                This will only work if the rule's ``trigger_type`` is :attr:`TriggerType.keyword`.
+        presets: Optional[List[:class:`KeywordPresetType`]]
+            The keyword presets that the filter should match.
+
+            .. note::
+                This will only work if the rule's ``trigger_type`` is :attr:`TriggerType.keyword_preset`
+        notify_channel: Optional[:class:`abc.GuildChannel`]
+            The channel that will receive the notification when this rule is triggered. Cannot be mixed with ``notify_channels``.
+        timeout_seconds: Optional[:class:`int`]
+            The seconds to timeout the person triggered this rule.
+        enabled: Optional[:class:`bool`]
+            Whether if this rule is enabled.
+        exempt_roles: Optional[List[:class:`Role`]]
+            A list of roles that should not be affected by this rule.
+        exempt_channels: Optional[List[:class:`abc.GuildChannel`]]
+            A list of channels that should not be affected by this rule.
+        """
+        payload = {}
+        if "name" in fields:
+            payload["name"] = fields["name"]
+
+        if "event_type" in fields:
+            payload["event_type"] = fields["event_type"].value
+
+        if "keyword_filters" in fields and self.trigger_type != TriggerType.keyword:
+            raise InvalidArgument(
+                "trigger_type must be TriggerType.keyword to pass keyword_filters"
+            )
+
+        if "keyword_filters" in fields:
+            payload["trigger_metadata"]["keyword_filters"] = fields["keyword_filters"]
+
+        if "presets" in fields and self.trigger_type != TriggerType.keyword_preset:
+            raise InvalidArgument("trigger_type must be TriggerType.keyword_preset to pass presets")
+
+        if "presets" in fields:
+            payload["trigger_metadata"]["presets"] = fields["presets"]
+
+        # TODO: notify_channel + timeout_seconds
```

Comment:
This pr is marked as ready for review, is this TODO complete or is the pr still under work?

Reply:
```suggestion
```